### PR TITLE
feat: custom numberpad

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Replaced AmountInput with custom number pad implementation

- add custom `NumberPad` with currency-specific types (simple, integer, decimal)
- `AmountInputViewModel` for centralized amount logic and state management
    - full Bitcoin/Fiat support with proper formatting and conversion
    - real-time max amount validation (999,999,999 sats) with haptic feedback
    - dynamic placeholders and smooth currency toggle animations
- refactor: `NumberPadTextField.swift`, `NumberPad.swift`, `ReceiveEdit.swift`
- add comprehensive edge case documentation added for E2E testing.

### Notes

Still need to add the custom number pad to some other flows, which may require some small refactoring. Testing all these edge cases will (hopefully) be covered soon by E2E tests.

### Screenshot / Video

Modern denomination:

https://github.com/user-attachments/assets/4f7e2143-06f8-4d9f-a8b6-e0d9566659f5

Classic denomination:

https://github.com/user-attachments/assets/bd12f7f3-ec70-45e5-8abd-193938245b99

### CODE REVIEW

File: Bitkit/Components/NumberPadTextField.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    var body: some View {
+        VStack(alignment: .leading, spacing: 16) {
+            // Secondary display (always shows the opposite of primary)
+            MoneyText(
+                sats: Int(viewModel.amountSats),
+                unitType: .secondary,
+                size: .caption,
+                symbol: true,
+                color: .textSecondary
+            )
```
Comment (jvsena42) [inline Bitkit/Components/NumberPadTextField.swift:20]: Non-blocking: In some screens, such as the Transfer flow, the secondary amount is not displayed. It could have a property like `showSecondaryAmount`

File: Bitkit/Docs/AmountInputRequirements.md
Comment (jvsena42) [inline Bitkit/Docs/AmountInputRequirements.md:1]: Nice!

**Review by ovitrif — APPROVED** [review]
LGTM :shipit:
